subgraph/README.md

```diff
@@ -0,0 +1,80 @@
+
+**Directories:**
+
+- [abis](./abis/): contains all ABIs related to all contracts of interest in the subgraph with their JSON format. These ABI files are all generated based on the verified version of the contract on the blockchain.
+- [build](./build/): contains the latest build of the subgraph with its WASM (which is ignored in the repository).
+- [generated](./generated/): contains the actual TypeScript implementation of the subgraph with the TS version of the graphql schema. The TS file in the `src` directory uses the TS implementation from here to function.
+- [src](./src/): contains the TypeScript implementation of the handlers of the subgraph for events. These handlers process every new event and store them in the database.
+- [tests](./tests/): contains the unit tests for this subgraph. For this purpose, `matchstick` (a unit testing framework by LimeChain) is used.
+- [query-examples](./query-examples): is the only directory that is not generated by the `graph-cli` package. You can find a working implementation of the `graphclient` along with a few queries there. For more information please check the [readme](./query-examples/README.md) for the examples.
+
+**Files:**
+
+- [networks.json](./networks.json): stores a list of the live versions of the contracts and their addresses on each chain they are deployed to.
+- [package.json](./package.json): a typical package.json file with commands to simplify working with the `graph-cli`.
+- [schema.graphql](./schema.graphql): the schema of all events, types, and structs we want to index in our subgraph.
+- [subgraph.yaml](./subgraph.yaml): the description of the subgraph with the list of events, paths to ABIs and the schema, information on the network of the subgraph, etc...
+- [tsconfig.json](./tsconfig.json): the TypeScript config file in the project.
+- [yarn.lock](./yarn.lock): Yarn's auto-generated lock file for keeping track of the exact versions of the packages that were used to run the code.
+
+## Build and deployment
+
+In order to deploy the subgraph, a live deployed instance of the contract is needed. If you are not already familiar with deploying contracts, you can follow the guide [here](https://github.com/fleekxyz/non-fungible-apps/tree/main#-deployment).
+
+When the contract is live, you can use the commands that are described in the `package.json` file to interact with the subgraph (build, deploy, test). But before doing so, please make sure you have the `graph-cli` package installed on your machine:
+
+```bash
+# If you want to use yarn:
+$ yarn global add @graphprotocol/graph-cli
+# If you want to use npm:
+$ npm install -g @graphprotocol/graph-cli
+```
+
+### Building the subgraph
+
+To build the subgraph, you can simply run the `yarn build` or `npm run build` commands. Under the hood, these commands use the `graph-cli` package to build the new schemas and config files of the subgraph. This is an essential step to do before deploying the subgraph.
+
```

Comment:
Maybe documenting the different schema and TS files would be useful in the readme